### ISSUE TITLE
OnClick property on FontAwesomeIcon typescript issue

### ISSUE THREAD

**Issue by e-osuna-g** [FortAwesome/react-fontawesome#196]
adding Onclick property to FontAwesomeIcon gives this error in the compiler.
example: 
`<FaIcon  onClick={this.showSettings} className={style.settingsIcon + ' ' +strIconClicked} icon="cog" />`
TS2339: Property 'onClick' does not exist on type 'IntrinsicAttributes & Props'.
i needed to wrap my <FaIcon in a span with the onClick, is there a way to add onClick and other Events to FontAwesomeIcon in typescript.

Thanks

**Comment by robmadole** [comment]
@cuauth can you provide a reproduction of this?

**Comment by robmadole** [comment]
How about a cloneable GitHub repo? That Stackblitz doesn't run locally.

**Comment by cuauth** [comment]
first time i run one of this, i know there's a lot of unused tools in the package json, but i took it from another project.
https://github.com/cuauth/FontAwesomeOnClick

**Comment by robmadole** [comment]
@cuauth I've release `0.1.4-1` that I think should fix this up. Can you give it a try in your real project?

**Comment by cuauth** [comment]
i'll hit it this weeked, and post updates, thanks

**Comment by kontinuity** [comment]
Can confirm this is not working on latest version.

```
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
...
<FontAwesomeIcon icon={faStar} size="1x" pull="right" onClick={() => message.success('Added to favorites!')} />
```

Results in:
```
TS2322: Type '{ icon: IconDefinition; size: "1x"; pull: "right"; onClick: () => MessageType; }' is not assignable to type 'IntrinsicAttributes & Props'.
  Property 'onClick' does not exist on type 'IntrinsicAttributes & Props'.
```

**Comment by Miikis** [comment]
Same as @kontinuity. A temporary fix is to wrap the FontAwesomeIcon in a container tag. 

```
<div onClick={onClick}>
    <FontAwesomeIcon icon={["fas", "heart"]} />
</div>
```

**Comment by gytisgreitai** [comment]
@robmadole any news on this?

**Comment by cuauth** [comment]
I haven't check it out but I will do it today if that's ok with you

**Comment by Plasmoxy** [comment]
Same here, I had to wrap the FontAwesomeIcon in <span onclick=...>  to make it clickable.

**Comment by peetjvv** [comment]
Thanks to @Plasmoxy for the tip, but it would be nice if we can get a working onClick prop. It's a little annoying to have to wrap all the icons in spans.

**Comment by DiegoGorgazzi** [comment]
I couldn't make onClick work no matter what I wrapped FontAwesomeIcon with. I was able to make it clickable ONLY when wrapping it with an anchor element, but kind of useless if onClick isn't working.

I ended up using unicode instead (I needed arrows) and fudged things via CSS.

**Comment by siffash** [comment]
@cuauth any news on this?

**Comment by elramus** [comment]
Yeah i get the same error here too, BUT -- are you sure you really want the onClick on the icon in the first place? Wrapping it in either a button or a link is the more accessibly correct way to do this, unless you're also adding aria stuff and tabindex to the icon too.

**Comment by siffash** [comment]
Yes, ultimately I ended up using a wrapper tag, BUT since in general it IS possible to add onClick to SVGs why do we need to restrict developers in the first place?

**Comment by kolendomichal** [comment]
Issue is still relevant. I had to wrap my **FontAwesomeIcon** in <span>

**Comment by fulin426** [comment]
Is wrapping the icon in a div or a span the only way to go around this issue? Without typescript I can put a onClick handler directly on the icon.

**Comment by siffash** [comment]
@fulin426 Please refer to the issue's name (it says "_typescript_ issue").

**Comment by fulin426** [comment]
@siffash Maybe I wasn't clear with my issue, it's essentially the same as the OP. Typescript is not allowing me to attach a onClick handler directly on the Fontawesome component.

**Comment by cuauth** [comment]
yeah thats the only workaround for typescript as far as i know.

**Comment by lwlewy** [comment]
Unfortunately, wrapping doesn't seem to work on IE11 :(

**Comment by JulienDemarque** [comment]
Maybe this issue should be re-opened. It's still throwing an error in Typescript.

**Comment by Tongboy** [comment]
why was this issue closed? it's still an issue with current 5.11.2

**Comment by BagdadiMan** [comment]
I think this issue should be re-opened, not `onClick` is not working, tries wrapping it with `span` and it still not working for me

**Comment by robmadole** [comment]
PRs are welcome on this issue. We don't use Typescript so it's going to be hard for us to know if we have fixed this or not. What about [this file](https://github.com/FortAwesome/react-fontawesome/blob/master/index.d.ts) needs to change?

**Comment by robmadole** [comment]
If anyone can provide a reproducible test case that we can run locally that would also help a lot.

**Comment by ihaseebkhan** [comment]
I think this issue is fixed as onClick is directly working on FontAwesomeIcon tag. <FontAwesomeIcon onClick={this.onClickHandler} icon={faShareAlt} />

**Comment by peetjvv** [comment]
I can confirm that it's working as expected for me now. I'm using v0.1.9

**Comment by kwent** [comment]
Same here works with `v0.1.11` ! Thank you

**Comment by achievermina** [comment]
I don't see onClick props in FontAwesomeIcon. 

"@fortawesome/fontawesome-svg-core": "^1.2.35",
"@fortawesome/free-brands-svg-icons": "^5.15.3",
"@fortawesome/free-solid-svg-icons": "^5.15.3",
"@fortawesome/react-fontawesome": "^0.1.14",

**Comment by robmadole** [comment]
@achievermina what do you mean that you don't see them? In you code editor?

**Comment by bartektartanus** [comment]
I can also confirm that it is working. Version `0.1.16`

**Comment by peetjvv** [comment]
yes, this issue can be closed
